### PR TITLE
[spirv] Support struct member functions

### PR DESCRIPTION
**Pull request by antiagainst**
Member functions are translated into SPIR-V standalone functions
with an additional first parameter for the struct type.

### CODE REVIEW

**Comment by antiagainst** [comment]
@ehsannas: PTAL

File: tools/clang/lib/SPIRV/SPIRVEmitter.cpp [hunk 1]
```diff
@@ -1601,180 +1655,173 @@ uint32_t SPIRVEmitter::doCXXMemberCallExpr(const CXXMemberCallExpr *expr) {
...
-      uint32_t bias = 0;
-      if (opcode == static_cast<uint32_t>(IntrinsicOp::MOP_SampleBias)) {
-        bias = doExpr(expr->getArg(2));
-      } else {
-        lod = doExpr(expr->getArg(2));
-      }
-      // .Bias()/.SampleLevel() has a fourth optional paramter for offset.
-      uint32_t constOffset = 0, varOffset = 0;
-      handleOffset(expr, 3, &constOffset, &varOffset);
+  switch (opcode) {
```
Comment (antiagainst) [inline tools/clang/lib/SPIRV/SPIRVEmitter.cpp]: FYI: There are no changes to the `switch` statement except that the `default` case is moved out.

**Comment by dmpots** [comment]
@antiagainst looks like there is a conflict from another merged pull request. I can merge after you resolve the conflicts.

**Comment by antiagainst** [comment]
Thanks @dmpots! I've rebased and now it's ready for merge.

**Pull request merged**